### PR TITLE
printutils OK, add `va_end` to variadic functions

### PR DESCRIPTION
**Pull request by EllipticEllipsis**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

Another small boot file bites the dust.

### CODE REVIEW

File: src/boot_O2/printutils.c [hunk 1]
```diff
@@ -1,5 +1,12 @@
...
+    _Printf(*pfn, pfn, fmt, args);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/boot/printutils/PrintUtils_Printf.s")
+void PrintUtils_Printf(PrintCallback* pfn, const char* fmt, ...) {
+    va_list args;
+    va_start(args, fmt);
+
+    PrintUtils_VPrintf(pfn, fmt, args);
+}
```
Comment (AngheloAlf) [inline src/boot_O2/printutils.c:12]: Shouldn't this kind of variadic functions have a `va_end` after it processed the arguments?
```suggestion
void PrintUtils_Printf(PrintCallback* pfn, const char* fmt, ...) {
    va_list args;
    va_start(args, fmt);

    PrintUtils_VPrintf(pfn, fmt, args);
    
    va_end(args);
}
```

**Comment by EllipticEllipsis** [comment]
Looks like all our variadic functions were missing `va_end`, so I've added them. Doesn't affect OK, probably at least partially because the IDO version of the macro is, well, empty, but may matter for more modern compilers.

**Review by AngheloAlf — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]
Looks good to me. Pending +1 contributor review.

**Review by jpburnett — APPROVED** [review]
looks good 👍

**Review by engineer124 — APPROVED** [review]
TIL how variadic functions work

**Pull request merged**